File: test_samples/ruby/user_service.rb
# frozen_string_literal: true

# Ruby service class demo
# Demo file for multi-language scanning

# Hardcoded secrets - should trigger UNI-001
API_KEY = "sk-live-abc123xyz789"  # TODO: env var
DB_PASSWORD = "postgres123"  # FIXME: rotate
AUTH_TOKEN = "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"
SECRET_KEY = "AES_SECRET_KEY_12345"

# Debug prints - should trigger UNI-002
puts "Starting Ruby service..."
pp config
warn "Warning: deprecated method"

# Deep nesting - should trigger UNI-005
def validate_input(input)
  if !input.nil?
    if !input.empty?
      if input.length > 0
        if input != ""
          if !input.nil?
            puts "Valid input: #{input}"
            return true
          end
        end
      end
    end
  end
  false
end

class UserService
  def initialize
    # Hardcoded secret
    @api_key = "sk-live-secret123"  # TODO: env
    puts "Initializing UserService..."
    puts "API_KEY: #{@api_key}"
  end

  def fetch_user(id)
    puts "Fetching user: #{id}"
    # Empty rescue - should trigger UNI-003
  rescue => e
    # empty rescue - BAD
  end

  def process_request(request)
    # Deep nesting
    if request
      if request[:params]
        if request[:params][:data]
          if request[:params][:data][:user]
            if request[:params][:data][:user][:name]
              puts "Processing: #{request[:params][:data][:user][:name]}"
            end
          end
        end
      end
    end
  end
end

service = UserService.new
service.fetch_user(123)

File: test_samples/typescript/user_service.ts
/**
 * TypeScript service class demo
 */

class UserService {
    // Hardcoded secrets - should trigger UNI-001
    private readonly API_URL = "https://api.example.com/v1";
    private readonly DB_PASSWORD = "MySecretPass123!";  // TODO: env var
    private readonly AWS_KEY = "AKIAIOSFODNN7EXAMPLE";  // FIXME: rotate key
    private readonly AUTH_TOKEN = "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9";

    // Deep nesting - should trigger UNI-005
    validateUser(user: any): boolean {
        if (user) {
            if (user.name) {
                if (user.name.length > 0) {
                    if (user.email) {
                        if (user.email.includes("@")) {
                            if (user.email !== "") {
                                console.log("Validating user:", user.email);
                                return true;
                            }
                        }
                    }
                }
            }
        }
        return false;
    }

    // Debug statements - should trigger UNI-002
    async fetchUser(id: string): Promise<any> {
        console.log("Fetching user:", id);
        console.log("API_URL:", this.API_URL);
        debugger;
        try {
            const response = await fetch(`${this.API_URL}/users/${id}`);
            // Empty catch - should trigger UNI-003
        } catch (e) {
            // do nothing
        }
        return response.json();
    }
}

export const userService = new UserService();
